### PR TITLE
feat: set the GDI-spec compliant defs and warn about missing service.name

### PR DESCRIPTION
**Pull request by Rauno56**
- feat: set the defaults on limits
- feat: warn about missing service.name and no instrumentations

# Description

Addresses some of the concerns from GDI spec review.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Internal change (a change which is not visible to the package consumers)
- [x] Added automated tests
- [x] Unit tests have been added/updated
- [x] Documentation has been updated
- [x] Change file has been generated (`npm run change:new`)
- [ ] Delete this branch (after the PR is merged)

### CODE REVIEW

File: src/options.ts [hunk 1]
```diff
@@ -94,12 +99,20 @@ export function _setDefaultOptions(options: Partial<Options> = {}): Options {
   const serviceName =
     options.serviceName ||
     process.env.OTEL_SERVICE_NAME ||
-    resource.attributes[SemanticResourceAttributes.SERVICE_NAME] ||
-    defaultServiceName;
+    resource.attributes[SemanticResourceAttributes.SERVICE_NAME];
+
+  if (!serviceName) {
```
Comment (seemk) [inline src/options.ts:104]: Is `diag.warn` visible in the terminal as well?

File: docs/advanced-config.md [hunk 2]
```diff
@@ -16,6 +16,13 @@ To configure the Splunk Distribution of OpenTelemetry JS, you can use a combinat
 
 ## List of settings
 
+This distribution supports all the configuration options supported by the components it uses with the defaults specified by the [OTel Specification](https://github.com/open-telemetry/opentelemetry-specification) with few exceptions:
+
+| Environment variable | Default value
+| --- | --- |
+| OTEL_ATTRIBUTE_VALUE_LENGTH_LIMIT | `1000` |
```
Comment (owais) [inline docs/advanced-config.md:23]: Should be 12000 https://github.com/signalfx/gdi-specification/blob/main/specification/configuration.md#opentelemetry-environment-variable

File: docs/advanced-config.md [hunk 3]
```diff
@@ -16,6 +16,13 @@ To configure the Splunk Distribution of OpenTelemetry JS, you can use a combinat
...
+| OTEL_SPAN_LINK_COUNT_LIMIT | `12000` |
```
Comment (owais) [inline docs/advanced-config.md:24]: should be `1000`

**Review by owais — APPROVED** [review]
Mistake in docs. Looks good otherwise.

File: docs/advanced-config.md (hunk 3 above)
Comment (Rauno56) [reply docs/advanced-config.md:24]: :D Thanks!

File: src/options.ts (hunk 1 above)
Comment (Rauno56) [reply src/options.ts:104]: I'll merge it, @seemk, but any comments are still welcome!

**Pull request merged**